### PR TITLE
JDK-8311009: Long.toUnsignedString(long, int) doesn't have to create a BigInteger

### PR DESCRIPTION
**Pull request by yuantj**
This PR changes the implementation of `Long.toUnsignedString(long, int)` for "default" radices, which avoids creating a `BigInteger` and makes use of `Long.divideUnsigned` and `Long.remainderUnsigned`.

I've run the test on `test/jdk/java/lang/Long/Unsigned.java` and it works correctly. I believe that there is no need to add more test cases.

I've added a benchmark case to `test/micro/org/openjdk/bench/java/lang/Longs.java` for this method. Here is the benchmark result tested on my machine:

Before (JDK 20.0.1):

```plain
Benchmark                       (size)  Mode  Cnt     Score    Error  Units
Longs.toUnsignedStringNegative     500  avgt   15  6428.711 ± 63.142  us/op
```

After:

```plain
Benchmark                       (size)  Mode  Cnt     Score     Error  Units
Longs.toUnsignedStringNegative     500  avgt   15  3823.655 ± 146.171  us/op
```

No CSR needed since the behavior is not changed.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311009](https://bugs.openjdk.org/browse/JDK-8311009): Long.toUnsignedString(long, int) doesn't have to create a BigInteger (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14654/head:pull/14654` \
`$ git checkout pull/14654`

Update a local copy of the PR: \
`$ git checkout pull/14654` \
`$ git pull https://git.openjdk.org/jdk.git pull/14654/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14654`

View PR using the GUI difftool: \
`$ git pr show -t 14654`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14654.diff">https://git.openjdk.org/jdk/pull/14654.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14654#issuecomment-1610640728)

### CODE REVIEW

**Comment by yuantj** [comment]
Please ignore the merge warning from the bot.

**Comment by rgiulietti** [comment]
The change looks good (but I'm not an official reviewer).
Please update the copyright year in the micro benchmark.

**Comment by yuantj** [comment]
BTW, do I need to remove `toUnsignedBigInteger` method (which is private and no longer used)?

**Comment by rgiulietti** [comment]
I checked whether the method is ever searched for reflectively: this seems _not_ be the case in the JDK codebase. So I guess it can be removed from `Long.java`.

File: src/java.base/share/classes/java/lang/Long.java [hunk 1]
```diff
@@ -234,7 +237,11 @@ public static String toUnsignedString(long i, int radix) {
                 }
                 case 16 -> toHexString(i);
                 case 32 -> toUnsignedString0(i, 5);
-                default -> toUnsignedBigInteger(i).toString(radix);
+                default -> {
+                    long leadingDigits = divideUnsigned(i, radix);  // always positive
+                    int lastDigit = (int)remainderUnsigned(i, radix);
```
Comment (rgiulietti) [inline src/java.base/share/classes/java/lang/Long.java:242]: You can avoid the additional division from `remainderUnsigned()` like so
```suggestion
                    int lastDigit = (int)(i - leadingDigits * radix);
```

**Pull request closed without merging**